File: 프로그래머스/lv0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.cpp
#include <string>
#include <vector>

using namespace std;


vector<string> solution(string my_str, int n)
{
    vector<string> answer;
    int size = 0;

    if (my_str.length() % n == 0)
        size = my_str.length() / n;
    else
        size = (my_str.length() / n)+1;
    
    int z = 0;
    for (int i = 0; i < size; i++)
    {
        answer.push_back(my_str.substr(z, n));
        z += n;
    }

    return answer;
}

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
class Solution {
    public String[] solution(String my_str, int n) {
        int size = (my_str.length() % n == 0) ? my_str.length() / n : my_str.length() / n + 1;
        String[] answers = new String[size];
        for (int i = 0, idx = 0; i < my_str.length(); i += n) {
            answers[idx++] = i + n > my_str.length() ? my_str.substring(i) : my_str.substring(i, i + n);
        }
        return answers;
    }
}
